### PR TITLE
Vishal's solution for Valid Parantheses

### PR DESCRIPTION
**Pull request by manideeppabba1991**

### CODE REVIEW

**Comment by manideeppabba1991** [comment]
Well Done on your first attempt! Let's not discuss the solution yet. We will dive through different concepts and revisit this solution

File: Java/Vishal/ValidParentheses.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      Output: true
+      
+  - Example 3:
+      Input: s = "(]"
+      Output: false
+       
+  
+  **Constraints**:
+  
+  1 <= s.length <= 104
```
Comment (RahulKakarla) [inline Java/Vishal/ValidParentheses.java:28]: this conditional check is missing

File: Java/Vishal/ValidParentheses.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                }
+                counter++;
+            }
+            if(numberOfOpenRoundParenthesis!=numberOfClosedRoundParenthesis
+                    || numberOfOpenFlowerParenthesis!=numberOfClosedFlowerParenthesis
+                    || numberOfOpenSquareParenthesis!=numberOfClosedSquareParenthesis) {
+                outPut=false;
+            }
+        }
+        return outPut;
```
Comment (RahulKakarla) [inline Java/Vishal/ValidParentheses.java:69]: didn't cover for order of occurrences  i guess, strings like this "}{" also will return true

File: Java/Vishal/ValidParentheses.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/*
+  Problem - Valid Parentheses
+  
+  Given a string s containing just the characters '(', ')', '{', '}', '[' and ']', determine if the input string is valid.
```
[the remark this answers is not in the document]
Comment (manideeppabba1991) [reply Java/Vishal/ValidParentheses.java:4]: I think he did. Constraints state that the string will consists of only above mentioned characters.

File: Java/Vishal/ValidParentheses.java (hunk 1 above)
Comment (manideeppabba1991) [reply Java/Vishal/ValidParentheses.java:28]: Constraints are the guidelines. This line means that given string will be of length between 1- 104.

File: Java/Vishal/ValidParentheses.java (hunk 2 above)
Comment (manideeppabba1991) [reply Java/Vishal/ValidParentheses.java:69]: Good catch! way to go!

File: Java/Vishal/ValidParentheses.java (hunk 1 above)
Comment (RahulKakarla) [reply Java/Vishal/ValidParentheses.java:28]: oh okay, my bad

File: Java/Vishal/BubbleSort.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//TimeComplexity = O(n^2)
+//SpaceComplexity = O(1)
+public void bubbleSort() {
+        int[] inputArray = new int[] {12,2,25,1,45,123,85};
+        for(int j=0; j<inputArray.length;j++) {
+            for(int i=0; i<inputArray.length-1; i++){
```
Comment (RahulKakarla) [inline Java/Vishal/BubbleSort.java:6]: every time we don't need to check till the end of the array, as we moved the biggest value to last place

**Review by RahulKakarla — APPROVED** [review]

**Pull request merged**